ShowerShow/Control/ShowerController.cs

```diff
@@ -11,45 +11,62 @@
+        private readonly IShowerService _showerService;
 
-        public ShowerController(ILogger<ShowerController> log)
+        public ShowerController(ILogger<ShowerController> log, ShowerService showerService)
         {
             _logger = log;
+            _showerService = showerService;
         }
-        [Function("CreateShower")]
-        [OpenApiOperation(operationId: "CreateShower", tags: new[] { "Showers" })]
-        [OpenApiRequestBody("application/json", typeof(CreateShowerDataDTO), Description = "The shower data.")]
+
+        [Function(nameof(GetShowerById))]
+        [OpenApiOperation(operationId: "getShowerById", tags: new[] { "Shower data" })]
+        [OpenApiParameter(name: "UserId", In = ParameterLocation.Path, Required = true, Type = typeof(Guid), Description = "The user id parameter")]
+        [OpenApiParameter(name: "ShowerId", In = ParameterLocation.Path, Required = true, Type = typeof(Guid), Description = "id of the requested shower")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(ShowerData), Description = "Successfully received the shower data.")]
+        public async Task<HttpResponseData> GetShowerById([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "user/{UserId}/showerdata/{showerId}")] HttpRequestData req, Guid UserId, Guid showerId)
+        {
+
+            var response = _showerService.GetShowerDataByUserId(UserId, showerId);
+
+            HttpResponseData responseData = req.CreateResponse(HttpStatusCode.OK);
+
+            await responseData.WriteAsJsonAsync(response);
+            return responseData;
+
+
+        }
+
+
+        [Function(nameof(CreateShower))]
+        [OpenApiOperation(operationId: "CreateShowerDataById", tags: new[] { "Shower data" })]
+        [OpenApiRequestBody("application/json", typeof(CreateShowerDataDTO), Description = "Created the shower data object")]
         [OpenApiParameter(name: "UserId", In = ParameterLocation.Path, Required = true, Type = typeof(Guid), Description = "The User ID parameter")]
         [OpenApiResponseWithBody(statusCode: HttpStatusCode.Created, contentType: "application/json", bodyType: typeof(ShowerData), Description = "The OK response with the new shower.")]
-        public async Task<IActionResult> CreateShower([HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "showerdata/{UserId:Guid}")] HttpRequestData req, Guid UserId)
+        public async Task<IActionResult> CreateShower([HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "user/{userId/showerdata")] HttpRequestData req, Guid UserId)
         {
             _logger.LogInformation("Creating new shower.");
 
             string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
 
             try
             {
-                CreateShowerDataDTO data = JsonConvert.DeserializeObject<CreateShowerDataDTO>(requestBody);
 
-                ShowerData shower = new();
```

Comment:
There is an automapper in utils you can use to autoconvert instead of doing it this way. Try to check it out